File: zipper_lists/zipper_lists.js
// class Node {
//   constructor(val) {
//     this.val = val;
//     this.next = null;
//   }
// }
​
const zipperLists = (head1, head2) => {
  let count = 0 // 2
  let res = head1 // B
  let curr1 = head1.next // C
  let curr2 = head2 // Y
​
  while (curr1 && curr2 != null) {
    if (count % 2 == 0) {
        res.next = curr2
        curr2 = curr2.next
    } else {
        res.next = curr1
        curr1 = curr1.next
    }
    res = res.next
    count++
  }
​
  if(curr1 === null) res.next = curr2
  if(curr2 === null) res.next = curr1
​
  return head1;
}
​
module.exports = {
  zipperLists,
};
​

File: zipper_lists/zipper_lists.py
# class Node:
#   def __init__(self, val):
#     self.val = val
#     self.next = None
​
def zipper_lists(head_1, head_2):
  res = head_1
  curr1 = head_1.next
  curr2 = head_2
  count = 0
​
  while curr1 and curr2 is not None:
    if count % 2 == 0:
      res.next = curr2
      curr2 = curr2.next
    else:
      res.next = curr1
      curr1 = curr1.next
    res = res.next
    count += 1
​
  if curr1 is None:
    res.next = curr2
  if curr2 is None:
    res.next = curr1
​
  return head_1
​